spec/association_spec.rb

```diff
@@ -570,6 +570,38 @@ class Tag < ActiveRecord::Base ; end
     end
   end
 
+  context 'manually defined through association' do
+    before(:each) do
+      create_tables(
+          "posts", {}, {},
+          "tags", {}, {},
+          "post_tags", {:id => false}, { :post_id => { foreign_key: true}, :tag_id => { foreign_key: true}}
+      )
+      class Post < ActiveRecord::Base
+        schema_associations concise_names: false do
+          has_many :linked_tags, through: :post_tags, source: :tag
+        end
+      end
+      class Tag < ActiveRecord::Base
+        schema_associations concise_names: false do
+          has_many :linked_posts, through: :post_tags, source: :post
+        end
+      end
+      class PostTag < ActiveRecord::Base
+      end
+    end
+
+    it 'should not error when accessing the through association' do
+      reflection = Post.reflect_on_association(:linked_tags)
+      expect(reflection).not_to be_nil
+      # expect(reflection.macro).to eq(:has_many)
+      # expect(reflection.options[:source]).to eq(:tag)
+      # expect(reflection.options[:through]).to eq(:posts_tags)
+
+      Post.new.linked_tags
```

Comment:
Maybe you'd want to assert something?

Reply:
yeah.. hadn't quite finished it.  It's done now.